app/controllers/stories_controller.rb

```diff
@@ -240,24 +240,35 @@ def assign_feed_stories
   end
 
   def assign_article_show_variables
+    not_found if permission_denied?
+    not_found unless @article.user
+
     @article_show = true
     @variant_number = params[:variant_version] || (user_signed_in? ? 0 : rand(2))
-    assign_user_and_org
+
+    @user = @article.user
+    @organization = @article.organization
+
+    if @article.collection
+      @collection = @article.collection
+
+      # we need to make sure that articles that were cross posted after their
+      # original publication date appear in the correct order in the collection,
+      # considering non cross posted articles with a more recent publication date
+      @collection_articles = @article.collection.articles.
+        published.
+        order("COALESCE(crossposted_at, published_at) ASC")
```

Comment:
This is the trick. we tell the DB to consider either `crossposted_at` or `published_at` in ascending order.

Reply:
`published_at` has an index, `crossposted_at` doesn't. I assume `@article.collection` will always be small enough that even in the presence of an index the query planner would prefer a seq scan, but I thought it's at least worth mentioning.

Reply:
Thanks for catching this! I'll make a note about investigating this and might add it in a future PR :)